### ISSUE TITLE
[Request] [Idea] Performance Recommendations

### ISSUE THREAD

**Issue by henry2man** [flutter/flutter-intellij#5484]
TL;DR: 

_As a developer -sometimes- I introduce performance problems in my applications_ 😅  **What if the Flutter plugin helps me detect and fix them?**

Full story: 

Reviewing some articles on recommendations to improve the performance of applications made with Flutter, I find some elements that could be added to the plugin to help developers improve the performance of their apps.

For example:
- It is recommended not to use methods that create Widgets (or lists of Widgets) within the `build()` method. Instead, specific Widgets must be created (https://www.youtube.com/watch?v=vVg9It7cOfY&t=1320s, https://blog.codemagic.io/how-to-improve-the-performance-of-your -flutter-app./#dont-split-your-widgets-into-methods)
- It is recommended to use lazy lists instead of creating all items manually (https://flutter.dev/docs/perf/rendering/best-practices#render-grids-and-lists-lazily)
- It is recommended not to use the `Opacity` widget (https://flutter.dev/docs/perf/rendering/best-practices#pitfalls)

I am not an absolute specialist in performance with Flutter but I have debugged on other platforms and it is a very important element in mobile apps.

My suggestion is that some of these items could appear as part of the warnings and/or as part of a specific performance analysis extension within the plugin.

What do you think?

**Comment by stevemessick** [comment]
@pq These sound like possible lint additions. What do you think?

**Comment by pq** [comment]
Excellent ideas @henry2man!  I'll transfer this issue over to the linter, which is where such analysis would happen.  Thanks!
